File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = []
    
    def func(arr, start):
        if not arr:
            if start == target: answer.append(1)
            return
        func(arr[1:], start+arr[0])
        func(arr[1:], start-arr[0])
    
    func(numbers, 0)
    return len(answer)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
import java.awt.*;

class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        Deque<Point> q = new ArrayDeque<>();
        q.offer(new Point(0, numbers[0]));
        q.offer(new Point(0, -numbers[0]));
        
        while (!q.isEmpty()) {
            Point now = q.poll();
            
            if (now.x == numbers.length - 1) {
                if (now.y == target) answer ++;
                continue;
            }
            
            for (int i = -1; i <= 1; i += 2) {
                if (now.x + 1 < numbers.length) {
                    int next = numbers[now.x+1] * i;
                    q.offer(new Point(now.x+1, now.y+next));
                }
            }
            
        }
        return answer;
    }
}